dspace-server-webapp/src/main/java/org/dspace/app/rest/ProcessRestController.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.linkTo;
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.methodOn;
+
+import java.sql.SQLException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dspace.app.rest.link.HalLinkService;
+import org.dspace.app.rest.link.process.ProcessResourceHalLinkFactory;
+import org.dspace.app.rest.model.BitstreamRest;
+import org.dspace.app.rest.model.ProcessRest;
+import org.dspace.app.rest.model.hateoas.BitstreamResource;
+import org.dspace.app.rest.model.hateoas.ProcessFileWrapperResource;
+import org.dspace.app.rest.repository.ProcessRestRepository;
+import org.dspace.app.rest.utils.Utils;
+import org.dspace.authorize.AuthorizeException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.web.PagedResourcesAssembler;
+import org.springframework.hateoas.Link;
+import org.springframework.hateoas.PagedResources;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/" + ProcessRest.CATEGORY + "/" + ProcessRest.PLURAL_NAME)
+public class ProcessRestController {
```

Comment:
Please rename this class...this is not a generic controller for the `processes` endpoint, as it looks like *most* of those requests would use `ProcessRestRepository`.  Instead, this should be updated to specifically deal with ` /processes/files` subresource.  So, maybe this should be:

`@RequestMapping("/api" + ProcessRest.CATEGORY + "/" + ProcessRest.PLURAL_NAME + "/files)`
`public class ProcessFilesRestController {`

Reply:
I think this Controller can now be removed in favor of a `LinkRestRepository` (see other `LinkRestRepository` classes as examples).  I don't see a reason why we this needs to a custom Controller

Reply:
We moved part of the controller to the new "ProcessFilesLinkRepository" class, the other ones couldn't. The LinkRestRepository framework does not allow us to further specify a repository than just the Category, Model and Name of the linkRest. This means that we’re stuck at the /files endpoint and can’t further specify path variables or further deeper URL structures. Therefore we need this Controller to take care of it.